### PR TITLE
Resolve most Doxygen warnings

### PR DESCRIPTION
**Pull request by joshessman-llnl**
The only significant remaining warnings are for the expression template operator overloads.  Any thoughts on whether there's value in documenting them?

### CODE REVIEW

File: src/common/expr_template_internal.hpp [hunk 1]
```diff
@@ -23,14 +23,21 @@
...
+ * @tparam vec The base vector expression type
+ * @tparam owns Whether the object "intends" to take ownership of the constructor argument
+ */
 template <typename vec, bool owns>
 using vec_t = typename std::conditional<owns, vec, const vec&>::type;
 
-// Type alias for the constructior argument to a vector expression - an
-// rvalue reference if ownership is desired (will be moved from), otherwise,
-// and lvalue reference
+/** Type alias for the constructior argument to a vector expression - an
```
Comment (white238) [inline src/common/expr_template_internal.hpp]: @brief ?

**Review by white238 — APPROVED** [review]
Thanks for taking care of this.

**Review by jamiebramwell — APPROVED** [review]
Thank you for doing this!

**Comment by jamiebramwell** [comment]
> 
> 
> The only significant remaining warnings are for the expression template operator overloads. Any thoughts on whether there's value in documenting them?

I think it's fine to leave these out.

**Comment by joshessman-llnl** [comment]
LGTM

**Pull request merged**